### PR TITLE
Recruit System (no menu yet)

### PR DESCRIPTION
**Pull request by ghost**
_Can't get the numbers to display, can't manage to extend the function arguments no matter how hard I try._

Edit: fixed thanks to Mr_Oinky.

### CODE REVIEW

File: src/engine/game/battle/enemybattler.lua [hunk 1]
```diff
@@ -618,10 +621,31 @@ function EnemyBattler:defeat(reason, violent)
...
+            counter.second_number = self.recruit
+            Assets.playSound("mercyadd", 4.8, 0.8) -- Is this the sounds that plays when an enemy gets recruited? I have no idea.
+        end
+        if self:getFlag("recruit", 0) >= self.recruit then
+            self:setFlag("recruit", true)
+        end
+    end
+    
     Game.battle.money = Game.battle.money + self.money
-    Game.battle.xp = Game.battle.xp + self.experience
```
Comment (NyakoFox) [inline src/engine/game/battle/enemybattler.lua:624]: While experience is an unused mechanic, it does get added here, and not on violence defeats.

File: src/engine/game/effects/recruitmessage.lua [hunk 2]
```diff
@@ -95,7 +94,7 @@ function RecruitMessage:draw()
 
     if (self.second_number > 1) then
         love.graphics.setFont(Assets.getFont("goldnumbers"))
-        love.graphics.print(tostring(self.first_number), self.texture:getWidth() - 70, 35)
+        love.graphics.print(tostring(self.first_number), self.texture:getWidth() - 70 - ((#tostring(self.first_number) - 1) * 20), 35)
```
Comment (NyakoFox) [inline src/engine/game/effects/recruitmessage.lua:97]: This is inaccurate. In Deltarune, the code is as follows:
```gml
draw_self()
if (secondnumber > 1)
{
    draw_set_color(c_white)
    draw_set_font(global.damagefontgold)
    draw_set_alpha(image_alpha)
    draw_text((x - 70), (y + 35), firstnumber)
    draw_text((x - 30), (y + 35), secondnumber)
    draw_sprite(spr_numbersfontbig_gold, 13, (x - 50), (y + 35))
    draw_set_alpha(1)
}
```

File: src/engine/game/battle/enemybattler.lua [hunk 3]
```diff
@@ -618,10 +621,31 @@ function EnemyBattler:defeat(reason, violent)
...
     end
-
+    
+    if self.recruit > 0 and type(self:getFlag("recruit", 0)) == "number" and (self.done_state == "PACIFIED" or self.done_state == "SPARED") then
+        self:addFlag("recruit", 1)
+        if Game:getConfig("enableRecruits") then
+            local counter = self:recruitMessage("recruit")
+            counter.first_number = self:getFlag("recruit", 0)
+            counter.second_number = self.recruit
+            Assets.playSound("mercyadd", 4.8, 0.8) -- Is this the sounds that plays when an enemy gets recruited? I have no idea.
```
Comment (NyakoFox) [inline src/engine/game/battle/enemybattler.lua:641]: It plays `sparkle_gem`.

**Review by NyakoFox — CHANGES_REQUESTED** [review]
First pass.

File: src/engine/game/battle/enemybattler.lua (hunk 1 above)
Comment (FireRainV) [reply src/engine/game/battle/enemybattler.lua:624]: Ralsei talked in Chapter 1 about sparing to not get EXP, if this mechanic ever gets used, it's probably if you cause a fatal kill. I don't think it matters that much for now.

File: src/engine/game/effects/recruitmessage.lua (hunk 2 above)
Comment (FireRainV) [reply src/engine/game/effects/recruitmessage.lua:97]: This is just an offset which fixes overlapping when the count is higher than 9. There're no enemies in Deltarune with that number.

File: src/engine/game/battle/enemybattler.lua (hunk 3 above)
Comment (NyakoFox) [reply src/engine/game/battle/enemybattler.lua:641]: Yes.

File: src/engine/game/battle/enemybattler.lua (hunk 1 above)
Comment (NyakoFox) [reply src/engine/game/battle/enemybattler.lua:624]: I'd rather not go off of guesses and just use what's currently in Deltarune. We can always change it later.

**Pull request merged**